### PR TITLE
docs: replace docs system

### LINKED ISSUE #439: Better documentation site
### What did you had in mind?

Replace `sphinx` with `mkdocs` which seems better for this project.

For refs, [this](https://switcher-webapi.tomfi.info/) is the docs site for this project build with `sphinx`,
and [this](https://archetypes.tomfi.info/) is a docs site built for another project of mine, build with `mkdocs`.

### Are you trying to fix a problem?

_No response_

### Any lead on how this feature can be implemented?

_No response_

### PR DESCRIPTION
**Pull request by dolby360**
Signed-off-by: dolby360 <dolevben054@gmail.com>

## Description
Replace sphinx with mkdocs
**Related issue (if any):** fixes #439 

## Checklist

- [x] I have followed this repository's contributing guidelines.
- [x] I have used the conventional commits specification for my commit messages.
- [x] I have signed my commits.
- [x] I will adhere to the project's code of conduct.

## Additional information

> Anything else?
>

### CODE REVIEW

**Comment by dolby360** [comment]
@TomerFi That's very interesting. auto-me-bot says the commit is unsigned.
I wonder why...

**Comment by TomerFi** [comment]
> @TomerFi That's very interesting. auto-me-bot says the commit is unsigned. I wonder why...

hmm... did you add the trailer manually or with git's signoff option?
because I think you're missing an empty line above the trailer.
can you please try adding it manually (git commit --amend)?

**Comment by dolby360** [comment]
> hmm... did you add the trailer manually or with git's signoff option? because I think you're missing an empty line above the trailer. 

No I didn't

> can you please try adding it manually (`git commit --amend`)?

Sure

**Comment by TomerFi** [comment]
That's weird...
I'm looking into it.

File: docs/apidocs.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# API documentations
```
Comment (TomerFi) [inline docs/apidocs.md:1]:
```suggestion
<!-- markdownlint-disable MD033 -->
# API documentations
```

Comment (TomerFi) [reply docs/apidocs.md:1]: this should resolve the lint ci failures.

**Comment by TomerFi** [comment]
@dolby360 
Thanks dude - I will merge this later - I just want to play with a bit more to figure out what's wrong with these ci failures.
For now, I skipped the dockerlint step in the lint pr workflow, and I will look into the signed commits workflow later, and merge this.

**Comment by dolby360** [comment]
@TomerFi Please let me know how can I help with the investigations.

**Review by TomerFi — APPROVED** [review]

**Comment by TomerFi** [comment]
@all-contributors add @dolby360  for doc

**Pull request merged**